src/site/content/en/patterns/media/microphone-process/assets/processor.js

```diff
@@ -0,0 +1,12 @@
+// This file is evaluated in the audio rendering thread
+// upon context.audioWorklet.addModule() call.
+
+class WorkletProcessor extends AudioWorkletProcessor {
+  process([input], [output]) {
+    // Copy inputs to outputs.
+    input[0].forEach((sample, i) => output[0][i] = sample);
```

Comment:
I think Float32Array.set() is much faster than this iteration - which is sort of the best practice for audio. What do you think?

Reply:
I also noticed that this doesn't store/copy the audio stream into something else. (e.g. recording) Is this pattern only to demonstrate access to raw audio data via AudioWorklet?

Reply:
> I think Float32Array.set() is much faster than this iteration - which is sort of the best practice for audio. What do you think?

Much better indeed. It is fixed in https://github.com/GoogleChrome/web.dev/pull/9198/commits/ecabb69725d5a2070e1a4f359a8d055cfcc0dce6. Thanks!

> I also noticed that this doesn't store/copy the audio stream into something else. (e.g. recording) Is this pattern only to demonstrate access to raw audio data via AudioWorklet?

This pattern is specifically about processing yes. We do https://web.dev/patterns/media/microphone-record/ for recording. I wonder now if we should have another pattern for processing & recording via AudioWorklet.

Reply:
Thanks for confirmation. LGTM!